receiver/dockerstatsreceiver/metric_helper.go

```diff
@@ -71,3 +75,47 @@ func calculateMemoryPercent(limit uint64, usedNoCache uint64) float64 {
 	}
 	return 0.0
 }
+
+// calculateCPULimit calculate the number of cpus assigned to a container.
+//
+// Calculation is based on 3 alternatives by the following order:
+// - nanocpus:   if set by i.e docker run -cpus=2
+// - cpusetCpus: if set by i.e docker run -docker run -cpuset-cpus="0,2"
+// - cpuquota:   if set by i.e docker run -cpu-quota=50000
+//
+// See https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler for background.
+func calculateCPULimit(hostConfig *ctypes.HostConfig) float64 {
+	var cpuLimit float64
+
+	switch {
+	case hostConfig.NanoCPUs > 0:
+		cpuLimit = float64(hostConfig.NanoCPUs) / 1e9
+	case hostConfig.CpusetCpus != "":
+		cpuLimit = parseCPUSet(hostConfig.CpusetCpus)
+	case hostConfig.CPUQuota > 0:
+		period := hostConfig.CPUPeriod
+		if period == 0 {
+			period = 100000 // Default CFS Period
+		}
+		cpuLimit = float64(hostConfig.CPUQuota) / float64(period)
+	}
+	return cpuLimit
+}
+
+// parseCPUSet helper function to decompose -cpuset-cpus value into number os cpus.
+func parseCPUSet(line string) float64 {
+	var numCPUs uint64
+
+	lineSlice := strings.Split(line, ",")
+	for _, l := range lineSlice {
+		lineParts := strings.Split(l, "-")
+		if len(lineParts) == 2 {
+			p0, _ := strconv.Atoi(lineParts[0])
```

Comment:
I wouldn't ignore any errors here. You never know. I would bubble them up and report them.

Reply:
Good point @gbbr. This should really be an edge case as the Docker API checks the syntax of `cpusetCpus` for invalid values, but still it won't hurt bubbling up the error to be safe